srtcore/core.cpp

```diff
@@ -8764,8 +8755,8 @@ int32_t CUDT::bake(const sockaddr *addr, int32_t current_cookie, int correction)
         // SYN cookie
         char clienthost[NI_MAXHOST];
         char clientport[NI_MAXSERV];
-        getnameinfo(addr,
-                    (m_iIPversion == AF_INET) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6),
+        getnameinfo(&addr,
+                    addr.size(),
```

Comment:
This expression is simply a shortcut to what was here before.

If you want to change the existing code behavior, I need to see why. Actually with your proposed change this function would fail.